### PR TITLE
bug: Should continue when predestroy fails with an execption

### PR DESCRIPTION
**Pull request by timyates**
[skip ci]

### CODE REVIEW

**Review by dstepanov — APPROVED** [review]

File: inject/src/main/java/io/micronaut/context/DefaultBeanContext.java [hunk 1]
```diff
@@ -1142,7 +1142,11 @@ private <T> void destroyBean(@NonNull BeanRegistration<T> registration, boolean
         beanToDestroy = triggerPreDestroyListeners(definition, beanToDestroy);
 
         if (definition instanceof DisposableBeanDefinition) {
-            ((DisposableBeanDefinition<T>) definition).dispose(this, beanToDestroy);
+            try {
+                ((DisposableBeanDefinition<T>) definition).dispose(this, beanToDestroy);
+            } catch (Exception e) {
+                LOG.info("Error disposing bean [{}]... Continuing...", beanToDestroy, e);
```
Comment (graemerocher) [inline inject/src/main/java/io/micronaut/context/DefaultBeanContext.java]: Should be `warn` and I don't think you can use parameterization like this to include the stack trace. Has to be normal string concatenation. Also include `e.getMessage()` in the error

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: inject/src/main/java/io/micronaut/context/DefaultBeanContext.java (hunk 1 above)
Comment (dstepanov) [reply inject/src/main/java/io/micronaut/context/DefaultBeanContext.java]: I think it will work, the last parameter can be the exception cause

**Review by sdelamo — APPROVED** [review]
@timyates please address @graemerocher request

File: inject/src/main/java/io/micronaut/context/DefaultBeanContext.java (hunk 1 above)
Comment (timyates) [reply inject/src/main/java/io/micronaut/context/DefaultBeanContext.java]: @dstepanov is right 😄 

Changed it to:

```
                if (LOG.isWarnEnabled()) {
                    LOG.warn("Error disposing bean [" + beanToDestroy.toString() + "]... Continuing...", e);
                }
```

**Comment by timyates** [comment]
@sdelamo done (hopefully)

**Pull request merged**